src/Detector/PlatformDetectorResult.cs

```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Oryx.Detector
 {
     /// <summary>
     /// Represents the result of a <see cref="IPlatformDetector.Detect(DetectorContext)"/> operation.
+    /// This contains basic information of the detected application. However, each individual detector could include
+    /// additional information about the application. For example <see cref="DotNetCore.DotNetCoreDetector"/> returns
+    /// an instance of <see cref="DotNetCore.DotNetCorePlatformDetectorResult"/>.
     /// </summary>
     public class PlatformDetectorResult
     {
+        /// <summary>
+        /// Gets or sets the name of the platform that was detected. For example: nodejs, dotnet, php and python.
```

Comment:
Thinking about this from an API perspective, these properties should be immutable - we only set them in the constructor and whoever got the object can't change them

Reply:
Ok, let me file a work item to track this. I will handle it in a separate PR.